File: gradle-plugin/src/main/java/juuxel/ripple/gradle/impl/SpecDependency.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package juuxel.ripple.gradle.impl;

import juuxel.ripple.gradle.DependencySpec;
import org.gradle.api.artifacts.Dependency;
import org.jetbrains.annotations.Nullable;

public interface SpecDependency extends Dependency {
    DependencySpec getSpec();

    @Nullable
    @Override
    default String getGroup() {
        return getSpec().getGroup();
    }

    @Override
    default String getName() {
        return getSpec().getName();
    }

    @Nullable
    @Override
    default String getVersion() {
        return getSpec().getVersion();
    }
}

File: gradle-plugin/src/main/kotlin/juuxel/ripple/gradle/dependency/SpecDependency.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package juuxel.ripple.gradle.dependency

import org.gradle.api.artifacts.Dependency

internal interface SpecDependency : Dependency {
    val spec: DependencySpec

    override fun getGroup() = spec.group
    override fun getName() = spec.name
    override fun getVersion() = spec.version
}
